### DISCUSSION TITLE
Unable to set retries or other arguments in non decorator task mode:

### DISCUSSION THREAD

**Discussion by JackDapid** [PrefectHQ/prefect#2954] [Q&A] [answered]
Does anybody have an example how to set task arguments with the non-task-decorator notation?

I have no issues using the max_retries or other task arguments when I use the task decorator:

```
@task(max_retries=10, retry_delay=datetime.timedelta(seconds=5))
def my_prefect_task():
...
```

but than if I try to do more complex prefect task classes:

```
class MyPrefectTask(Task(max_retries=10, retry_delay=datetime.timedelta(seconds=5))):
    def __init__(self, df=None, **kwargs):
...
```

I get some binding errors:
```
ValueError: Could not infer an active Flow context.
/usr/local/lib/python3.7/site-packages/prefect/core/task.py:416: ValueError

```

**Comment by joshmeek** [comment, marked as answer]
@JackDapid Try setting the task arguments at initialization like this:

```python
class MyPrefectTask(Task):
    def __init__(self, df=None, **kwargs):
        ....

my_task = MyPrefectTask(max_retries=10, retry_delay=datetime.timedelta(seconds=5))
```

**Reply by JackDapid** [reply]
@joshmeek, thanks for the super fast reaction, yes that works, but is rather uncomfortable in the context I am using it in.

While in decorator @task(...) case I am setting the 'default' retries for a task and than create as many instances as I need with these settings, with your proposed solution I would have to set these arg every time (or at every location) instead of having them centrally with the definition of the task. But maybe I naively assumed that setting of the default args like in the @task(...) decorator notation would also work in the Task() class notation - I will have a look what the @task decorator actually does as I assumed it would turn a normal function into a Task() class

Or is there another way to set the default retries, name, ... arguments?
🤔

**Reply by JackDapid** [reply]
https://github.com/PrefectHQ/prefect/blob/1f43f5acaab6c5bacea48bd23d8b5e37a875e2fd/src/prefect/utilities/tasks.py#L318
->
https://github.com/PrefectHQ/prefect/blob/6f1c8e1d2f9fef31596e8565342bc252e21322bd/src/prefect/tasks/core/function.py#L27

**Reply by joshmeek** [reply]
I believe you want something like this:

```python
class MyPrefectTask(Task):
    def __init__(self, df=None, max_retries=10, **kwargs):
        ....
        super().__init__(max_retries=max_retries, **kwargs)

my_task = MyPrefectTask(retry_delay=datetime.timedelta(seconds=5))  # using default
my_task = MyPrefectTask(max_retries=10, retry_delay=datetime.timedelta(seconds=5))  # not using default
```

**Reply by JackDapid** [reply]
THX @joshmeek yes, that was it 👍

**Reply by cicdw** [reply]
FYI: you can technically achieve this without classes as well:

```python
@task(max_retries=10, retry_delay=datetime.timedelta(seconds=5))
def my_prefect_task():
    pass


# within a Flow context
my_prefect_task(task_args={max_retries=5}) # overrides `max_retries`
```

I think @joshmeek 's suggestion is cleaner / easier to reason about but wanted to point this out for reference

**Comment by JackDapid** [comment]
Thanks @cicdw, yes I have used the decorator notation @task for that successfully before, I use the class notation for other reasons, but I totally agree fro others that have the same questions your comment will be very helpfull, thank you
